enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,465 @@
+
+Once we have the detected devices the administrator can create the localPVs by explicitly selecting the disks. This can be done by the `localVolume` CR. The other option would be to create localPVs via the AutoDetectVolume CR and passing the inclusion filters in it.
+
+
+#### Workflow of LocalPV creation via LocalVolume CR
+
+- Discovery: The user can choose to run discovery if they want to understand what devices are available across the nodes in the cluster.
+    - ##### NOTE: This step is optional for users directly creating CRs. They might already know what CRs they need for step 2.
+- PV Creation: After the user has decided how to created PVs, they can choose to create PVs with either the LocalVolume CR or the AutoDetectVolume CR.
+
+
+## Design Details for `Auto discovery of local devices`
+
+API scheme for `LocalVolumeDiscovery` and `LocalVolumeNodeDiscoveryResult` CR:
+
+```go
+type LocalVolumeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeDiscovery `json:"items"`
+}
+
+type LocalVolumeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeDiscoverySpec   `json:"spec"`
+	Status            LocalVolumeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoverySpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscoveryStatus struct {
+	Phase         string `json:"phase,omitempty"`
+	CompletedJobs *int32 `json:"completedjobs,omitempty"`
+	TotalJobs     *int32 `json:"totaljobs,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryResultList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeNodeDiscoveryResult `json:"items"`
+}
+
+type LocalVolumeNodeDiscoveryResult struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeNodeDiscoveryResultSpec   `json:"spec"`
+	Status            LocalVolumeNodeDiscoveryResultStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryResultSpec struct {
+	NodeName string `json:"nodeName"`
+}
+
+type LocalVolumeNodeDiscoveryResultStatus struct {
+	Phase      string `json:"phase,omitempty"`
+	TimeStramp string `json:"timestramp,omitempty"`
+	// Blockdevices will be containing the output of following command unmarshalled into the structured
+	// "lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint"
+	Blockdevices []Device `json:"blockdevices"`
```

Comment:
`BlockDevices`

Reply:
renamed it to just devices.

Reply:
beacuse we will also be discovering the partitions